### PR TITLE
fix(sdk): update define function names in Next.js compat

### PR DESCRIPTION
**Pull request by heypoom**
We renamed the type helper functions to `defineMetabaseTheme` and `defineMetabaseAuthConfig` in https://github.com/metabase/metabase/pull/50585. This PR updates the compat generation script to use the same naming.

### CODE REVIEW

**Review by npretto — APPROVED** [review]

File: enterprise/frontend/src/embedding-sdk/bin/generate-nextjs-compat.ts [hunk 1]
```diff
@@ -51,7 +50,6 @@ const COMPONENTS_TO_EXPORT: ComponentDefinition[] = [
 
 // END OF CONFIGURATION
 
-// eslint-disable-next-line no-literal-metabase-strings -- it's code
 const MetabaseProviderCode = `
 const MetabaseProvider = ({
   config,
```
Comment (npretto) [inline enterprise/frontend/src/embedding-sdk/bin/generate-nextjs-compat.ts:55]: We also need to update these

**Review by npretto — CHANGES_REQUESTED** [review]
I was testing this locally, it wasn't working because we also renamed `config` -> `authConfig`

**Review by npretto — APPROVED** [review]
Tested locally, works as expected

**Comment by heypoom** [comment]
@metabase-bot backport release-x.51.x

**Pull request merged**